specification/transport/udp/udp.tex

```diff
@@ -0,0 +1,337 @@
+# Incremented by one for each subsequent frame of a multi-frame transfer.
+
+bool end_of_transfer
+# True if this is the last frame of a multi-frame transfer, or a single-frame transfer.
+
+uint16 user_data
+# Opaque application-specific data with user-defined semantics.
+# Generic implementations should emit zero and ignore this field upon reception.
+
+uint8[2] header_crc16_big_endian
+# CRC-16/CCITT-FALSE of the preceding serialized header data in the big endian byte order.
+# Application of the CRC function to the entire header shall yield zero, otherwise the header is malformed.
+
+@assert _offset_ / 8 == {24}
+@sealed     # The payload data follows.
+\end{minted}
+\end{samepage}
+
+The header CRC function is \textbf{CRC-16/CCITT-FALSE};
+refer to section~\ref{sec:appendix_crc16ccitt_false} for further information.
+
+\begin{remark}
+    Certain states provided in the header duplicate information that is already available in the IP header
+    or the multicast group address.
+    This is done for reasons of unification of the header format with other standard transport layer definitions,
+    and to simplify the access to the transfer parameters that otherwise would be hard to reach above the
+    network layer, such as the DSCP value.
+    The latter consideration is particularly important for forwarding nodes.
+\end{remark}
+
+\subsection{Transfer payload}
+
+After the transfer payload is constructed but before it is scheduled for transmission over the network,
+it is appended with the transfer CRC field.
+The transfer CRC function is \textbf{CRC-32C} (section~\ref{sec:appendix_crc32c}),
+and its value is serialized in the little-endian byte order.
+The transfer CRC function is applied to the entire transfer payload and only transfer payload.
+
+The transfer CRC is provided for all transfers,
+including single-frame transfers and transfers with an empty payload\footnote{%
+    This provides end-to-end integrity protection for the transfer payload.
+}.
+An implementation receiving a transfer should verify the correctness of its transfer CRC.
+
+\begin{remark}
+    From the perspective of the multi-frame segmentation logic, the transfer CRC field is part of the transfer payload.
+    From the definition of the header format it follows that the transfer CRC can only be found at the end of
+    the packet if the \verb|end_of_transfer| bit is set,
+    unless the transfer CRC field has spilled over to the next frame
+    (in which case the frame would contain only the transfer CRC itself or the tail thereof).
+\end{remark}
+
+\subsection{Maximum transmission unit}
+
+In this section, the maximum transmission unit (MTU) is defined as the maximum size of a UDP/IP datagram payload.
+
+This specification does not restrict the MTU of the underlying transport.
+It is recommended, however, to avoid MTU values less than 508~bytes to allow reduced (simplified) implementations
```

Comment:
How about:

By setting MTU values greater than or equal to 508 bytes, any system that does not define messages larger than 480 bytes ($508 - 24 - 4$) can simplify their implementation by omitting support for multi-frame transfers.